minecraft_script/compiler/compile_types.py

```
@@ -93,6 +141,30 @@ def __repr__(self) -> str:
+    def __init__(self, context, component: dict | None = None):
+        super().__init__(context, "text_component")
+        self.component = dict(component or {})
+        from .text_component_builtins import attach_text_component_methods
+
+        attach_text_component_methods(self)
+
+    def clone(self) -> "MCSTextComponent":
+        cloned = MCSTextComponent(self.context, deepcopy(self.component))
+        return cloned
+
+    def save_to_storage_cmd(self, value: any = None) -> str:
```

Ruff (0.15.15)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)